src/main/java/io/github/thebusybiscuit/slimefun4/core/config/SlimefunConfigManager.java

```diff
@@ -0,0 +1,275 @@
+            } catch (Exception x) {
+                item.error("Something went wrong while updating the permission node for this item!", x);
+                isSuccessful = false;
+            }
+        }
+
+        return isSuccessful;
+    }
+
+    /**
+     * This method returns whether the {@link Server} should be restarted to apply the latest changes.
+     * 
+     * @return Whether a restart is required.
+     */
+    public boolean isRestartRequired() {
+        return isRestartRequired;
+    }
+
+    @Nonnull
+    public Config getPluginConfig() {
+        return pluginConfig;
+    }
+
+    @Nonnull
+    public Config getItemsConfig() {
+        return itemsConfig;
+    }
+
+    @Nonnull
+    public Config getResearchConfig() {
+        return researchesConfig;
+    }
+
+    /**
+     * This method saves all our {@link Config} files.
+     */
+    public void saveFiles() {
+        pluginConfig.save();
+        itemsConfig.save();
+        researchesConfig.save();
+    }
+
+    /**
+     * This method returns whether backwards-compatibility is enabled.
+     * Backwards compatibility allows Slimefun to recognize items from older versions but comes
+     * at a huge performance cost.
+     * 
+     * @return Whether backwards compatibility is enabled
+     */
+    public boolean isBackwardsCompatible() {
+        return isBackwardsCompatibilityEnabled;
+    }
+
+    /**
+     * This method sets the status of backwards compatibility.
+     * Backwards compatibility allows Slimefun to recognize items from older versions but comes
+     * at a huge performance cost.
+     * 
+     * @param compatible
+     *            Whether backwards compatibility should be enabled
+     */
+    public void setBackwardsCompatible(boolean compatible) {
+        isBackwardsCompatibilityEnabled = compatible;
+    }
+
+    public void setResearchingEnabled(boolean enabled) {
+        isResearchingEnabled = enabled;
+    }
+
+    public boolean isResearchingEnabled() {
+        return isResearchingEnabled;
+    }
+
+    public void setFreeCreativeResearchingEnabled(boolean enabled) {
+        isResearchingFreeInCreativeMode = enabled;
+    }
+
+    public boolean isFreeCreativeResearchingEnabled() {
+        return isResearchingFreeInCreativeMode;
+    }
+
+    public boolean isResearchFireworkEnabled() {
+        return isResearchFireworkEnabled;
+    }
+
+    public boolean isDuplicateBlockLoggingEnabled() {
+        return isDuplicateBlockLoggingEnabled;
+    }
+
+    public boolean isVanillaRecipeShown() {
+        return isVanillaRecipeShowingEnabled;
+    }
+
+    public boolean isSlimefunGuideGivenOnJoin() {
+        return isSlimefunGuideGivenOnJoin;
+    }
+
+    public boolean isUpdaterEnabled() {
+        return isUpdaterEnabled;
+    }
+
+    public boolean isTalismanMessageInActionbar() {
+        return isTalismanMessageInActionbar;
+    }
+
+    public boolean isExcessCommandItemsDroppingEnabled() {
```

Comment:
these methods SHOULD have JavaDocs but I'll let you off ;)

Reply:
sigh, later :P